sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Manager.RestClient.cs

```diff
@@ -101,18 +102,30 @@ private void OnPost()
                 {
                     _lastSuccessfulPost = DateTimeOffset.UtcNow;
 
+                    bool etagUpdated = false;
                     if (response.ConfigurationEtag != null && response.ConfigurationEtag != _etag)
                     {
                         Debug.WriteLine($"OnPost: updated etag: {response.ConfigurationEtag}");
                         _etag = response.ConfigurationEtag;
+                        etagUpdated = true;
                     }
 
-                    if (response.Subscribed)
+                    if (!response.Subscribed)
                     {
                         Debug.WriteLine($"OnPost: Subscribed: {response.Subscribed}");
                         _etag = string.Empty;
                         SetPingState();
+                        // TODO: double check if subscribedValue is false, we should stop collecting properly: (QuickPulseTelemetryModule.OnStopCollection).
+                    }
+                    else
+                    {
+                        if (etagUpdated && response.CollectionConfigurationInfo != null)
+                        {
+                            _collectionConfigurationInfo = response.CollectionConfigurationInfo;
```

Comment:
I think you can remove the `etagUpdated` bool if you move the `response.CollectionConfigurationInfo != null` check into the if block at line 106.

Reply:
If I move it there, the `_collectionConfigurationInfo` might be updated without checking whether `response.Subscribed` is true. Even though server shouldn't send us contradictory data like this.

I've rearranged the check order a bit: we check `Subscribed` value first, then does the `response.CollectionConfigurationInfo` check inside the `else` block.